server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -0,0 +1,110 @@
+/** *************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.mail
+
+import java.util.Objects
+
+import org.apache.james.core.Username
+import org.apache.james.jmap.model.UnsignedInt
+import org.apache.james.mailbox.Role
+import org.apache.james.jmap.model.Id
+import org.apache.james.mailbox.model.MailboxId
+
+final case class MailboxName(name: String) {
+  require(name != null, "'name' is mandatory")
+  require(!name.isEmpty, "'name' is mandatory")
+}
+
+final case class MailboxRights(mayReadItems: Boolean,
+                               mayAddItems: Boolean,
+                               mayRemoveItems: Boolean,
+                               maySetSeen: Boolean,
+                               maySetKeywords: Boolean,
+                               mayCreateChild: Boolean,
+                               mayRename: Boolean,
+                               mayDelete: Boolean,
+                               maySubmit: Boolean)
+
+object MailboxNamespace {
+  sealed case class Type(value: String)
+
+  val Delegated = Type("Delegated")
+  val Personal = Type("Personal")
+
+  def delegated(owner: Username): MailboxNamespace = {
+    require(owner != null)
+    require(!owner.asString.trim.isEmpty)
```

Comment:
remove the null checking and can the userName use refined.
Otherwise do you have to throw couldn't  you return  an Either or at least a Try ?